test/e2e/scripts/run-instance/10-setup-kind.sh

```diff
@@ -3,7 +3,7 @@
 set -euo pipefail
 
 download_and_install_kubectl() {
-    curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+    curl --retry 5 -LO "https://dl.k8s.io/release/$(curl --retry 5 -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
```

Comment:
From https://github.com/curl/curl/issues/6712, this will only catch transfer errors, not 4xx / 5xx response errors. Adding `--fail` would be required to make the command fail on 4xx / 5xx response errors.

However, with `--fail`, 4xx / 5xx are not retried by default - you need to also use the `--retry-all-errors` option, which is only available from `curl 7.71.0` onwards.

Reply:
Let's give it a try and see if it works